src/app/progress/Raids.tsx

```diff
@@ -28,24 +29,28 @@ export default function Raids({
   defs: D2ManifestDefinitions;
   profileInfo: DestinyProfileResponse;
 }) {
-  const allMilestones: DestinyMilestone[] =
-    profileInfo.characterProgressions &&
-    profileInfo.characterProgressions.data &&
-    profileInfo.characterProgressions.data[store.id]
-      ? Object.values(profileInfo.characterProgressions.data[store.id].milestones)
-      : [];
+  const profileMilestoneData = idx(
+    profileInfo,
+    (p) => p.characterProgressions.data[store.id].milestones
+  );
+  const allMilestones: DestinyMilestone[] = profileMilestoneData
+    ? Object.values(profileMilestoneData)
+    : [];
 
   // filter to milestones with child activities of type <ActivityType "Raid" 2043403989>
   const filteredMilestones = allMilestones.filter((milestone) => {
-    const def = defs && defs.Milestone.get(milestone.milestoneHash);
+    const milestoneActivities = idx(
+      defs.Milestone.get(milestone.milestoneHash),
+      (m) => m.activities
+    );
```

Comment:
`idx` seems unnecessary here.

Reply:
there was a much better way to one-line this. thanks!